### ISSUE TITLE
resolv role fails in system which does not have systemd-resolved installed

### ISSUE THREAD

**Issue by iay** [arillso/ansible.system#36]
I tried to use your `arillso.system.resolv` role in one of my systems, as it seemed to be very close to what I needed. It failed because the role unconditionally stops and disables the `systemd-resolved` service, which doesn't happen to be installed on the system, which is a Debian 12 installation.

I don't know whether systems like that are in scope for your collection, or if you intend that its roles can be used in isolation like this, but if so then I guess it might be worth considering parameterising that behaviour.

**Comment by sbaerlocher** [comment]
@iay Thanks for the feedback on the `arillso.system.resolv` issue. I'll look into it and plan to implement a flexible solution through a pull request as soon as I have time

**Comment by sbaerlocher** [comment]
@iay I've made some adjustments to the arillso.system.resolv role to address the issue you mentioned. Specifically, I added a task to gather information about all services at the beginning of the playbook, which allows me to check for the presence of the systemd-resolved service before attempting to stop and disable it. Additionally, I modified the condition for the task that stops and disables systemd-resolved to ensure it only runs if the service is actually present. These enhancements should make the role more flexible and compatible with systems like Debian 12, where systemd-resolved might not be installed. Thank you for bringing this issue to my attention, and I hope these changes improve the role's usability in your environment.
